crates/cext/src/circuit.rs

```diff
@@ -152,22 +152,25 @@ pub unsafe extern "C" fn qk_circuit_free(circuit: *mut CircuitData) {
 /// @param circuit A pointer to the circuit to add the gate to.
 /// @param gate The StandardGate to add to the circuit.
 /// @param qubits The pointer to the array of ``uint32_t`` qubit indices to add the gate on. This
-///     can be a null pointer if there are no qubits for `gate` (e.g. `QkGate_GlobalPhase`)
+///     can be a null pointer if there are no qubits for ``gate`` (e.g. ``QkGate_GlobalPhase``).
 /// @param params The pointer to the array of ``double`` values to use for the gate parameters.
-///     This can be a null pointer if there are no parameters for `gate` (e.g. `QkGate_H`).
+///     This can be a null pointer if there are no parameters for ``gate`` (e.g. ``QkGate_H``).
+///
+/// @return An exit code.
 ///
 /// # Example
 ///
 ///     QkCircuit *qc = qk_circuit_new(100);
-///     qk_circuit_gate(qc, QkGate_H, *[0], *[]);
+///     uint32_t qubit[1] = {0};
+///     qk_circuit_gate(qc, QkGate_H, qubit, NULL);
```

Comment:
_Technically_ I think `&{0U}` is valid C99 syntax that would produce a pointer to an anonymous struct type that would decay (maybe with an extra cast) to a pointer to `uint32_t` on any platform where `unsigned int` is 32-bits (which is everything we support), but yeah, what you wrote is probably better haha.

Fwiw, you can write
```c
uint32_t qubit = 0;
qk_circuit_gate(qc, QKGate_H, &qubit, NULL);
```
if you like - you don't actually need an array object if you only need a pointer to a single integer (same principle as how Rust's `slice::from_ref` can work).  I don't think you can apply `&` to an integer literal (unlike Rust, which does a little bit of magic to make it work, if it can't inline the call and remove the reference altogether).